server/controllers/UserController.js

```diff
@@ -86,28 +86,59 @@ const update = (req, res, next) => {
                     logger.warn(`[404] PATCH /user userName: ${userName} - UserController: User not found`)
                     return
                 }
-                User.findOneAndUpdate({ "userName": userName }, updateData)
-                    .then(response => {
-                        // Success (200)
-                        res.status(200).json({
-                            message: `User ${userName} updated successfully!`,
-                            displayName: updateData.displayName,
-                            description: updateData.description
+
+                // Check if the achievement is locked (first-time update)
+                if (user.achievements[0].locked) {
+                    // First-time update, unlock achievement 1
+                    user.achievements[0].locked = false;
+                    user.save()
+                        .then(updatedUser => {
+                            // Success (200)
+                            res.status(200).json({
+                                message: `User ${userName} updated successfully and earned Achievement 1!`,
+                                displayName: updateData.displayName,
+                                description: updateData.description
+                            });
+                            logger.info(`[200] PATCH /user userName: ${userName} - UserController: Update user successful and earned Achievement 1`);
+                            logger.cont(`displayName: ${updateData.displayName}`);
+                            logger.cont(`description: ${updateData.description}`);
```

Comment:
Maybe we can create a separate function to check for achievements? 
This update function is getting really long and I think would be easier for us to maintain it with function calls for example: `updateAchievements()` which will call `updateDescriptionAchievement()`